lib/resty/openssl/revoked.lua

```diff
@@ -0,0 +1,85 @@
+local ffi = require "ffi"
+local bn_lib = require("resty.openssl.bn")
+require("resty.openssl.include.x509.crl")
+require("resty.openssl.include.x509.revoked")
+local NID_crl_reason = 141
+local C = ffi.C
+local ffi_gc = ffi.gc
+local _M = {}
+local revoked_ptr_ct = ffi.typeof('X509_REVOKED *')
+
+--- Creates new instance of X509_REVOKED data
+-- @tparam bn|number sn Serial number as number or bn instance
+-- @tparam number time Revocation time
+-- @tparam number reason Revocation reason
+-- @treturn table instance of the module or nil
+-- @treturn[opt] string Returns optional error message in case of error
+function _M.new(sn, time, reason)
+    --- only convert to bn if it is number
+    if type(sn) == "number"then
+        sn = bn_lib.new(sn)
+    end
```

Comment:
let's make sure otherwise `sn` is a `bn` instance, otherwise error out

Reply:
Done